examples/structured_data/advanced_techniques_for_structured_data.py

```diff
@@ -0,0 +1,472 @@
+raw_data.head()
+
+"""
+The two catrogircal features in the dataset are binary encoded.
+We will convert this dataset repesentation to the typical repesentation, where each categorical
+feature is repesented as on column.
+"""
+
+soil_type_values = [f"soil_type_{idx+1}" for idx in range(40)]
+wilderness_area_values = [f"area_type_{idx+1}" for idx in range(4)]
+
+soil_type = raw_data.loc[:, 14:53].apply(
+    lambda x: soil_type_values[0::1][x.to_numpy().nonzero()[0][0]], axis=1
+)
+wilderness_area = raw_data.loc[:, 10:13].apply(
+    lambda x: wilderness_area_values[0::1][x.to_numpy().nonzero()[0][0]], axis=1
+)
+
+CSV_HEADER = [
+    "Elevation",
+    "Aspect",
+    "Slope",
+    "Horizontal_Distance_To_Hydrology",
+    "Vertical_Distance_To_Hydrology",
+    "Horizontal_Distance_To_Roadways",
+    "Hillshade_9am",
+    "Hillshade_Noon",
+    "Hillshade_3pm",
+    "Horizontal_Distance_To_Fire_Points",
+    "Wilderness_Area",
+    "Soil_Type",
+    "Cover_Type",
+]
+
+data = pd.concat(
+    [raw_data.loc[:, 0:9], wilderness_area, soil_type, raw_data.loc[:, 54]],
+    axis=1,
+    ignore_index=True,
+)
+data.columns = CSV_HEADER
+
+# Change the target label indicies to range from 0 to 6.
+data["Cover_Type"] = data["Cover_Type"] - 1
+
+print(f"Dataset shape: {data.shape}")
+data.head().T
+
+"""
+You can see now the Dataframe has 13 columns per sample (12 features, plus the target label).
+
+Let's split the data into training and testing splits, with 85% and 15% of the instance, respectively.
+"""
+
+train_data, test_data = train_test_split(
+    data, train_size=0.85, stratify=data["Cover_Type"]
+)
+
+print(f"Train split size: {len(train_data.index)}")
+print(f"Test split size: {len(test_data.index)}")
+
+"""
+Now we store the train and test data splits to CSV files.
+"""
+
+train_data_file = "train_data.csv"
+test_data_file = "test_data.csv"
+
+train_data.to_csv(train_data_file, index=False)
+test_data.to_csv(test_data_file, index=False)
+
+"""
+## Define dataset metadata
+"""
+
+TARGET_FEATURE_NAME = "Cover_Type"
+
+TARGET_FEATURE_LABELS = ["0", "1", "2", "3", "4", "5", "6"]
+
+NUMERIC_FEATURE_NAMES = [
+    "Aspect",
+    "Elevation",
+    "Hillshade_3pm",
+    "Hillshade_9am",
+    "Hillshade_Noon",
+    "Horizontal_Distance_To_Fire_Points",
+    "Horizontal_Distance_To_Hydrology",
+    "Horizontal_Distance_To_Roadways",
+    "Slope",
+    "Vertical_Distance_To_Hydrology",
+]
+
+CATEGORICAL_FEATURES_WITH_VOCABULARY = {
+    "Soil_Type": list(data["Soil_Type"].unique()),
+    "Wilderness_Area": list(data["Wilderness_Area"].unique()),
+}
+
+CATEGORICAL_FEATURE_NAMES = list(CATEGORICAL_FEATURES_WITH_VOCABULARY.keys())
+
+FEATURE_NAMES = NUMERIC_FEATURE_NAMES + CATEGORICAL_FEATURE_NAMES
+
+COLUMN_DEFAULTS = [
+    [0] if feature_name in NUMERIC_FEATURE_NAMES + [TARGET_FEATURE_NAME] else ["NA"]
+    for feature_name in CSV_HEADER
+]
+
+NUM_CLASSES = len(TARGET_FEATURE_LABELS)
+
+"""
+Second, let's download the train and test data files
```

Comment:
Suggestion: 

- Since "First" wasn't mention much much earlier, let's refactor
- Use "training" (set) for consistency

```suggestion
In this section, you'll download the training and test data:
```